unit-tests/internal/internal-tests-extrinsic.cpp

```diff
@@ -92,3 +154,262 @@ TEST_CASE("Extrinsic graph management", "[live][multicam]")
+                for (auto profile : res.second)
+                {
+                    auto fps = profile.fps;
+                    if (device_type == "D400" && profile.stream == RS2_STREAM_ACCEL) fps = 250;
+                    cfg.enable_stream(profile.stream, profile.index, profile.width, profile.height, profile.format, fps); // all streams in cfg
+                    cfg_size += 1;
+                }
+
+                for (auto it = new_frame.begin(); it != new_frame.end(); it++)
+                {
+                    it->second = 0;
+                }
+                auto start_time = std::chrono::system_clock::now().time_since_epoch();
+                auto start_time_milli = std::chrono::duration_cast<std::chrono::milliseconds>(start_time).count();
+                bool condition = false;
+                std::mutex mutex;
+                std::mutex mutex_2;
+                auto process_frame = [&](const rs2::frame& f)
+                {
+                    std::lock_guard<std::mutex> lock(mutex_2);
+                    auto stream_type = f.get_profile().stream_name();
+                    auto frame_num = f.get_frame_number();
+                    auto time_of_arrival = f.get_frame_metadata(RS2_FRAME_METADATA_TIME_OF_ARRIVAL);
+
+                    if (std::find(frame_number[stream_type].begin(), frame_number[stream_type].end(), frame_num) == frame_number[stream_type].end())
+                    {
+                        if (!new_frame[stream_type])
+                        {
+                            frame_number[stream_type].push_back(frame_num);
+                            streams_delay[stream_type].push_back(time_of_arrival - start_time_milli);
+                            new_frame[stream_type] = true;
+                        }
+                        new_frame[stream_type] += 1;
+                    }
+                };
+                auto frame_callback = [&](const rs2::frame& f)
+                {
+                    std::lock_guard<std::mutex> lock(mutex);
+                    if (rs2::frameset fs = f.as<rs2::frameset>())
+                    {
+                        // With callbacks, all synchronized stream will arrive in a single frameset
+                        for (const rs2::frame& ff : fs)
+                        {
+                            process_frame(ff);
+                        }
+                    }
+                    else
+                    {
+                        // Stream that bypass synchronization (such as IMU) will produce single frames
+                        process_frame(f);
+                    }
+                };
+                if (is_pipe)
+                {
+                    rs2::pipeline_profile profiles = pipe.start(cfg, frame_callback);
+                }
+                else {
+                    for (auto& s : res.first)
+                    {
+                        start_time = std::chrono::system_clock::now().time_since_epoch();
```

Comment:
Why it is resampled ?

Reply:
start_time should be for each sensor (unlike pipe)